lib/_http_incoming.js

```diff
@@ -315,6 +315,8 @@ IncomingMessage.prototype._dump = function _dump() {
   if (!this._dumped) {
     this._dumped = true;
     this.resume();
+    // Flush the existing buffered data
+    this._readableState._flush();
```

Comment:
@mcollina yes, it works. However, it won't work for the code that call `req.abort()` and followed by a `res.on('data', cb);`. Without flush the buffered data or explicitly suppress
 the 'data' event , there is the chance that someone who registers the 'data' event later gets notifications.

It's not perfect but I am fine with the change if it impacts less